mutable_tree.go

```diff
@@ -567,6 +575,30 @@ func (tree *MutableTree) deleteVersionsFrom(version int64) error {
 	return nil
 }
 
+// deleteNodes deletes all nodes which have greater version than current, because they are not useful anymore
+func (tree *MutableTree) deleteNodes(version int64, hash []byte) {
+	if len(hash) == 0 {
+		return
+	}
```

Comment:
I don't think this is correct. We should skip `nil`, but empty trees will have a `[]byte{}` root node, and these should be removed as well. Would be great with a test-case for this.

Reply:
Empty root doesn't present as a node in DB so it can be skipped.

Reply:
Moreover tree.ndb.GetNode will panic with zero-length hash

Reply:
That seems to be the case. There _is_ an `[]byte{}` entry for the empty root in the database, but it appears to be correctly removed when calling `LoadVersionForOverwriting`.